Src/Sarif.PatternMatcher.Sdk/StaticValidatorBase.cs

```diff
@@ -1,15 +1,48 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
 
 using Microsoft.RE2.Managed;
 
 namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk
 {
     public abstract class StaticValidatorBase : ValidatorBase
     {
-        public IEnumerable<ValidationResult> IsValidStatic(IDictionary<string, FlexMatch> groups)
+        /// <summary>
+        /// Examines the groups output by a positive regex match and
+        /// optionally performs additional validated to determine
+        /// whether the match is valid.
+        /// </summary>
+        /// <param name="groups">
+        /// The named groups resulting from a positive regex match. This
+        /// data also transports a certain number of injected well-known
+        /// named values (such as the file name of the scan target).
+        /// </param>
+        /// <param name="perFileFingerprintCache">
+        /// A cache of file + fingerprint combinations that have been
+        /// observed previously. Our scanner will only detect and validate
+        /// a unique fingerprint once per file. Many regexes will produce
+        /// multiple matches that resolve to the same unique credential in
+        /// a file (one of the perils of multiline regex matching). It is
+        /// possible that this cache may drop the location of a second
+        /// actual match that happens to be duplicated in the file. In
+        /// practice, we will not worry about this scenario: it will be
+        /// sufficient that we flag one instance of the unique secret.
+        ///
+        /// The type of this cache is peculiar. We use a concurrent dictionary
```

Comment:
>We use a concurrent dictionary [](http://example.com/codeflow?start=48&length=30)

Here we say we are using a ConcurrentDictionary with <string, byte> but, in code, we are using an ISet<string>.

Looking at the AnalyzeContext, we are using a HashSet<string>.

Can you update the text or switch back to concurrentDictionary?

Reply:
'DRAFT ONLY: don't review' :)